simple-mdns/src/lib.rs

```diff
@@ -53,15 +53,15 @@ fn create_socket(addr: &SocketAddr) -> io::Result<Socket> {
     Ok(socket)
 }
 
-fn join_multicast(addr: &SocketAddr) -> io::Result<UdpSocket> {
+fn join_multicast(addr: &SocketAddr, interface: &Ipv4Addr) -> io::Result<UdpSocket> {
```

Comment:
You are right, an enum would be possible there. I will change that later.

Reply:
I was just reading the code again, I've realized that the Ipv6 functionality is not exposed and everything is hardcoded to use ipv4.

If you prefer, you can rollback the change to the way it was before, since there is no way for the user to choose the ipv6 interface today